src/commands/moderation.rs

```diff
@@ -155,55 +155,61 @@ pub async fn unlock(ctx: Context<'_>) -> Result<(), Error> {
     .await
 }
 
-/// Unmute a member.
+/// Unmute a user.
 #[poise::command(slash_command)]
 pub async fn unmute(
     ctx: Context<'_>,
-    #[description = "The member to unmute"] member: Member,
+    #[description = "The user to unmute"] user: UserId,
 ) -> Result<(), Error> {
+    let user = to_user!(user, ctx);
+    let id = user.id;
     ctx.defer().await.expect("Failed to defer");
 
     let data = &ctx.data().read().await;
     let configuration = &data.configuration;
 
-    if let Some(pending_unmute) = data.pending_unmutes.get(&member.user.id.0) {
-        trace!("Cancelling pending unmute for {}", member.user.id.0);
+    if let Some(pending_unmute) = data.pending_unmutes.get(&id.0) {
+        trace!("Cancelling pending unmute for {}", id.0);
         pending_unmute.abort();
     }
 
     let author = ctx.author();
 
     let queue = queue_unmute_member(
-        &ctx.discord().http,
-        &data.database,
-        &member,
+        ctx.discord().clone(),
+        data.database.clone(),
+        ctx.guild_id().unwrap(),
```

Comment:
Why not pass `ctx` to `queue_unmute_member`?

Reply:
> Why not pass `ctx` to `queue_unmute_member`?

I am already passing a **serenity** context. I cannot make `queue_unmute_member` accept a **poise** context because we do not have access to that inside event handlers.